keel-core/src/main/kotlin/com/netflix/spinnaker/keel/api/Resource.kt

```diff
@@ -48,6 +48,9 @@ data class SubmittedResource<T : Any>(
   val spec: T
 )
 
+val <T : ResourceSpec> SubmittedResource<T>.name: ResourceName
+  get() = "${apiVersion.prefix}:$kind:${spec.name}".let(::ResourceName)
```

Comment:
This replaces `generateName` in the handlers. Much simpler. Only works on a properly parsed spec.